src/local/room_name.rs

```diff
@@ -153,6 +155,55 @@ impl RoomName {
     }
 }
 
+impl Into<JsValue> for RoomName {
+    fn into(self) -> JsValue {
+        let name = self.to_array_string();
+
+        JsValue::from_str(name.as_str())
+    }
+}
+
+impl Into<JsValue> for &RoomName {
+    fn into(self) -> JsValue {
+        let name = self.to_array_string();
+
+        JsValue::from_str(name.as_str())
+    }
+}
+
+impl Into<JsString> for RoomName {
+    fn into(self) -> JsString {
+        let val: JsValue = self.into();
+
+        val.unchecked_into()
+    }
+}
+
+impl Into<JsString> for &RoomName {
+    fn into(self) -> JsString {
+        let val: JsValue = self.into();
+
+        val.unchecked_into()
+    }
+}
+
+impl From<JsValue> for RoomName {
```

Comment:
This and below should really be `TryFrom`, seeing as they could very easily fail.

Reply:
I'll move them to be an 'unchecked_into' since they're only intended to be used on the marshalling boundary and visible to this crate.

Reply:
I made them try_from but added 'expect' unwrapping in places it should never fail.